source/encoding.c

```diff
@@ -499,6 +506,13 @@ int aws_utf8_validator_update(struct aws_utf8_validator *validator, struct aws_b
                 }
             }
         }
+
+        // Extra parse for caller defined validation callback.
```

Comment:
Why wouldn't this go up in the if statement from 488-500?

Reply:
Because it need also take care of the case for 1 byte codepoint (ln 459 - 463). Or I can add it to both place instead of here so we can reduce the branching.